packages/fleet/lib/utils/url.ts

```diff
@@ -0,0 +1,21 @@
+import type { Result } from '@nangohq/utils';
+import { Err, Ok } from '@nangohq/utils';
+import { setTimeout } from 'node:timers/promises';
+
+export async function waithUntilHealthy({ url, timeoutMs }: { url: string; timeoutMs: number }): Promise<Result<void>> {
+    const startTime = Date.now();
+    const waitMs = 1000;
+    while (Date.now() - startTime < timeoutMs) {
+        try {
+            const res = await fetch(url);
```

Comment:
Hopefully, nodejs does not cache dns failure that could slow down this check (but I think they are notà